packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -3,52 +3,326 @@ pragma solidity 0.8.11;
+   * @param supportedAsset - The address of the whitelisted asset. If the native asset is to be whitelisted,
+   * the address of the wrapped version will be stored
+   * @param caller - The account that called the function
+   */
+  event AssetAdded(bytes32 canonicalId, uint32 domain, address adoptedAsset, address supportedAsset, address caller);
+
   /**
-   * @notice Gets unique identifier from nonce + domain
-   * @param _nonce - The nonce of the contract
-   * @param _params - The call params of the transfer
-   * @return The transfer id
+   * @notice Emitted when an asset is removed from whitelists
+   * @param canonicalId - The canonical identifier of the token removed
+   * @param caller - The account that called the function
    */
-  function getTransferId(
-    uint256 _nonce,
-    address _sender,
-    IConnext.CallParams calldata _params
-  ) external pure returns (bytes32) {
-    return keccak256(abi.encode(_nonce, _sender, _params));
+  event AssetRemoved(bytes32 canonicalId, address caller);
+
+  /**
+   * @notice Emitted when a rlayer is added or removed from whitelists
+   * @param relayer - The relayer address to be added or removed
+   * @param caller - The account that called the function
+   */
+  event RelayerAdded(address relayer, address caller);
+
+  /**
+   * @notice Emitted when a rlayer is added or removed from whitelists
+   * @param relayer - The relayer address to be added or removed
+   * @param caller - The account that called the function
+   */
+  event RelayerRemoved(address relayer, address caller);
+
+  /**
+   * @notice Emitted when `reconciled` is called by the bridge on the destination domain
+   * @param transferId - The unique identifier of the crosschain transaction
+   * @param origin - The origin domain of the transfer
+   * @param routers - The CallParams.recipient provided, created as indexed parameter
+   * @param asset - The asset that was provided by the bridge
+   * @param amount - The amount that was provided by the bridge
+   * @param caller - The account that called the function
+   */
+  event Reconciled(
+    bytes32 indexed transferId,
+    uint32 indexed origin,
+    address[] routers,
+    address asset,
+    uint256 amount,
+    address caller
+  );
+
+  /**
+   * @notice Emitted when a router withdraws liquidity from the contract
+   * @param router - The router you are removing liquidity from
+   * @param to - The address the funds were withdrawn to
+   * @param local - The address of the token withdrawn
+   * @param amount - The amount of liquidity withdrawn
+   * @param caller - The account that called the function
+   */
+  event LiquidityRemoved(address indexed router, address to, address local, uint256 amount, address caller);
+
+  /**
+   * @notice Emitted when `xcall` is called on the origin domain
+   */
+  event XCalled(
+    bytes32 indexed transferId,
+    IConnext.XCallArgs xcallArgs,
+    xCalledEventArgs args,
```

Comment:
aren't these event args a bit redundant here?

`XCallArgs` includes `transactingAssetId` and `amount` already.

`XCalledArgs` seems to have only 2 unique vars defined `bridgedAmt` and `bridged`... what are those vars referring to exactly?

Reply:
To illustrate, here's everything that's being emitted in this event:
```
transferId, xcallArgs(params(to, callData, originDomain, destinationDomain), transactingAssetId, amount, relayerFee), args(transactingAssetId, amount, bridgedAmt, bridged), nonce, message, caller
```

(notice `transactingAssetId` and `amount` appear twice)

Reply:
Maybe the names are redundant? but the values could be different.
e.g.
- xcallArgs. transactingAssetId could be `address(0)` so args.transactingAssetId would be `address(wrapper)` (https://github.com/connext/nxtp/blob/poc-connext-router/packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol#L720-L722)
- amount could be different if the token being transfer is a deflationary ERC20 (https://github.com/connext/nxtp/blob/poc-connext-router/packages/deployments/contracts/contracts/lib/Connext/AssetLogic.sol#L75)

Reply:
looks good! yeah layne explained these are indeed not redundant. we'll have to fix up names later